### PR TITLE
sun3: Add DIAG/NORMAL switch

### PR DESCRIPTION
**Pull request by eschaton**
Adds a DIAG/NORMAL switch that affect how a Sun-3/60 (and other models) are configured.

### CODE REVIEW

**Comment by eschaton** [comment]
I used "Diagnostic" and "Normal" for the values; `DIAG` and `NORM` is how the hardware is labeled. Which is preferable for MAME?

File: src/mame/sun/sun3.cpp [hunk 1]
```diff
@@ -809,6 +814,23 @@ void sun3_state::vmetype3space_map(address_map &map)
...
+}
+
+void sun3_state::enable_w(uint32_t data, uint32_t mem_mask)
+{
+	// Mask out diag switch value, it comes from the switch on read.
+	const uint32_t diag = 0x01000000;
+	const uint32_t diag_norm_mask = ~diag;
+	COMBINE_DATA(&m_enable);
+	m_enable &= diag_norm_mask;
+}
```
Comment (cuavas) [inline src/mame/sun/sun3.cpp:832]: You can still end up with an issue doing it this way if someone injects a value with the bit set via a debugger memory viewer or something.  You’re better off masking the value on read – something like:

```c++
uint32_t sun3_state::enable_r()
{
	// Incorporate diag switch value.
	const uint32_t diagsw = m_diagsw->read() << 24;
	return m_enable & ~(u32(1) << 24) | diagsw;
}
```

and don’t bother with `enable_w` at all.

File: src/emu/ioport.cpp [hunk 2]
```diff
@@ -235,6 +235,7 @@ const struct
 	{ INPUT_STRING_Reverse, "Reverse" },
 	{ INPUT_STRING_Alternate, "Alternate" },
 	{ INPUT_STRING_None, "None" },
+	{ INPUT_STRING_Diagnostic, "Diagnostic" },
```
Comment (cuavas) [inline src/emu/ioport.cpp:238]: Please don’t add an input type for this.  Just use `PORT_CONFNAME` etc or `PORT_BIT( ..., ..., IPT_OTHER ) PORT_TOGGLE PORT_NAME(...)`

Comment (eschaton) [reply src/emu/ioport.cpp:238]: Aha! I only did it this way because I hadn't figured that out. :)

**Pull request merged**